### PR TITLE
[tomlplusplus] Migrate build to CMake

### PR DESCRIPTION
**Pull request by chris1786**
Migrates the build of tomlplusplus to CMake from meson.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by chris1786** [comment]
> There should not be a `meson.build` file under the `include/` directory.
> image

@MonicaLiu0311 - I'll look at this properly later today, but there isn't currently a way to stop this file being installed (I suspect it's being installed by mistake). I think this leaves two options:

1) Apply a patch within the call to `vcpkg_from_github()`
2) Remove the file after `vcpkg_cmake_install()` has been called

I'm favouring option two, as it's simpler to implement. What is your preference?

**Comment by MonicaLiu0311** [comment]
> > There should not be a `meson.build` file under the `include/` directory.
> > image
> 
> @MonicaLiu0311 - I'll look at this properly later today, but there isn't currently a way to stop this file being installed (I suspect it's being installed by mistake). I think this leaves two options:
> 
> 1. Apply a patch within the call to `vcpkg_from_github()`
> 2. Remove the file after `vcpkg_cmake_install()` has been called
> 
> I'm favouring option two, as it's simpler to implement. What is your preference?

I choose 2, just remove it, thank you.

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
tomlplusplus provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(tomlplusplus CONFIG REQUIRED)
    target_link_libraries(main PRIVATE tomlplusplus::tomlplusplus)
```

**Review by MonicaLiu0311 — APPROVED** [review]

File: ports/tomlplusplus/portfile.cmake [hunk 1]
```diff
@@ -6,20 +6,11 @@ vcpkg_from_github(
...
-        -Dgenerate_cmake_config=true
-        -Dbuild_tests=false
-        -Dbuild_examples=false
-)
-
-vcpkg_install_meson()
-vcpkg_cmake_config_fixup(CONFIG_PATH share/cmake/tomlplusplus)
-cmake_path(NATIVE_PATH SOURCE_PATH native_source_path)
-vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/tomlplusplus/tomlplusplusConfig.cmake" "${native_source_path}" "")
-vcpkg_fixup_pkgconfig()
```
Comment (Neumann-A) [inline ports/tomlplusplus/portfile.cmake:21]: Simply no, breaks pkg-config usage.
This removes pc files from the port without a replacement.

**Review by Neumann-A — CHANGES_REQUESTED** [review]

**Comment by Neumann-A** [comment]
@MonicaLiu0311 Please always check the file list before and after. This removes upstream provided pc files from the output! meson is the main upstream build system. The CMake build does not generate pc files.

**Comment by MonicaLiu0311** [comment]
I will turn this PR into a draft status, waiting for a response.

**Comment by MonicaLiu0311** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**